### PR TITLE
Upgrade sidekick installer

### PR DESCRIPTION
**Pull request by jxstxn1**
- Upgraded Sidekick Plugin Installer to `^0.2.1`
- Upgraded install file
- Moved `tool/build_ios_command.dart` to `template/build_ios_command.template.dart`
- Formatted and fixed analyzer

### CODE REVIEW

File: tool/install.dart [hunk 1]
```diff
@@ -2,22 +2,16 @@ import 'package:sidekick_core/sidekick_core.dart'
...
-  }
+  addSelfAsDependency();
   pubGet(package);
 
   final commandFile =
       package.root.file('lib/src/commands/build_ios_command.dart');
-  commandFile.writeAsStringSync(buildIosCommand);
+  PluginContext.installerPlugin.root
+      .file('template/build_ios_command.template.dart')
+      .copy(commandFile.path);
```
Comment (passsy) [inline tool/install.dart:14]: Prefer it in two lines 😉

Comment (jxstxn1) [reply tool/install.dart]: I made it smaller but i think im far from where you want it right? 😄

File: tool/install.dart [hunk 2]
```diff
@@ -2,22 +2,14 @@ import 'package:sidekick_core/sidekick_core.dart'
...
-  }
+  addSelfAsDependency();
   pubGet(package);
 
-  final commandFile =
-      package.root.file('lib/src/commands/build_ios_command.dart');
-  commandFile.writeAsStringSync(buildIosCommand);
+  PluginContext.installerPlugin.root
+      .file('template/build_ios_command.template.dart')
+      .copy(package.root.file('lib/src/commands/build_ios_command.dart').path);
```
Comment (passsy) [inline tool/install.dart:12]: Caution! `copy` is a Future and not awaited here

**Comment by passsy** [comment]
Works :)

**Pull request merged**